### PR TITLE
UW-24112 Update eslint npm dependencies

### PR DESCRIPTION
**Pull request by urbnjamesmi1**
Ticket: https://urbnit.atlassian.net/browse/UW-24112

Doing as suggested in https://github.com/urbn/SlipStream/pull/4370#pullrequestreview-518171699

- Update eslint to version 7
- Fix linting errors

### CODE REVIEW

File: package.json
Comment (brophdawg11) [inline package.json:38]: Should we bump all the `eslint*` deps in tandem to ensure they all play nice together?  The vue one is a major bump so unsure if that brings about new rules that cause breakages for us or not?  If it does, we can hold off on that and do it separately

```
eslint                                              6.8.0           6.8.0        7.12.1  vue-ssr-build
eslint-config-airbnb-base                          14.0.0          14.0.0        14.2.0  vue-ssr-build
eslint-plugin-import                               2.20.0          2.20.0        2.22.1  vue-ssr-build
eslint-plugin-vue                                   6.1.2           6.1.2         7.1.0  vue-ssr-build
```

And then we can bump the version of this to the next beta release and I'll tag once merged

Comment (urbnjamesmi1) [reply package.json]: Done

Comment (brophdawg11) [reply package.json]: I tried a _very_ quick install of this branch over in SlipStream and got the following error trying to lint.  I'm going to try a more thorough `npm ci` though and see if it was maybe just a package mismatch from a different branch:

```
Oops! Something went wrong! :(

ESLint: 7.12.1

TypeError: Cannot read property 'value' of null
Occurred while linting /Users/brophym1/urbn/repos/slipstream/.storybook/preview.js:143
    at checkSpacingBefore (/Users/brophym1/urbn/repos/slipstream/node_modules/eslint/lib/rules/template-curly-spacing.js:52:24)
    at TemplateElement (/Users/brophym1/urbn/repos/slipstream/node_modules/eslint/lib/rules/template-curly-spacing.js:136:17)
    at /Users/brophym1/urbn/repos/slipstream/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    ...
```

Just want to make sure that we don't need a subsequent change (and thus a new beta release) after we merge/tag this

Comment (urbnjamesmi1) [reply package.json]: I followed the solution steps given in https://github.com/babel/babel/issues/10904 and pushed that change here.  Now on master of SlipStream, I have 167 linting errors (mostly coming from test and storybook files).  Should I push up those fixes under another UW-24112 PR?

**Review by brophdawg11 — APPROVED** [review]

File: package.json
Comment (brophdawg11) [reply package.json]: yeah - let me see what they look like and we can see if they're rules we want to enforce or not also.  LEt's get this merged and then I'll bump to beta.5 and we can open a PR in slipstram to point to that and see what the errors look like

**Pull request merged**
